05_concurrency/homework/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type Cache interface {
+	Set(key string, value interface{}, ttl time.Duration)
+	Get(key string) (interface{}, bool)
+	Delete(key string)
+}
+
+type Item struct {
+	value            interface{}
+	ttl              time.Duration
+	invalidationTime int64
+}
+
+type CacheImplementation struct {
+	sync.Mutex
+	items map[string]Item
```

Comment:
Using this structure from another package is impossible because it does not have constructor (`NewCache` or something like that) and `items` field is unexported, so I cannot create it like

```
import "daniil/cache"

c := &cache.CacheImplementation{}
```
because its map will be nil internally.

This could be fixed by checking if map of key-values is initialized inside `Set`.